tests/alg_comp.c

```diff
@@ -41,6 +42,60 @@ const struct {
     {}
 };
 
+const uint32_t long_string_tests[] = {
+    200000, MAX_COMPRESSION_SIZE, 300000, 0
+};
+
+static uint8_t* get_random_string(uint32_t length)
+{
+    assert(length);
+    uint8_t* c = (uint8_t*)malloc(length*sizeof(uint8_t));
+    for (uint32_t i=0; i<length; i++) {
+        c[i] = 'A' + (random() % 26);
+    }
+    return c;
+}
+
+static void
+test_long_string(size_t inputlen) {
+    jose_io_auto_t *b = NULL;
+    jose_io_auto_t *c = NULL;
+    jose_io_auto_t *z = NULL;
+    void *buf1 = NULL;
+    void *buf2 = NULL;
+    size_t blen = 0;
+    size_t clen = 0;
+    const jose_hook_alg_t *a = jose_hook_alg_find(JOSE_HOOK_ALG_KIND_COMP, "DEF");
+    uint8_t* str = get_random_string(inputlen);
+
+    /* Test compression first. */
+    b = jose_io_malloc(NULL, &buf1, &blen);
+    assert(b);
+    z = a->comp.def(a, NULL, b);
+    assert(z);
+
+    if(inputlen > MAX_COMPRESSION_SIZE) {
+        assert(!z->feed(z, str, inputlen));
```

Comment:
it seem you are testing the limit only on compression, where the problem is really on decompression of untrusted input.

Ideally you use the bare compression code to created a valid, but larger than MAX_COMPRESSED_SIZE buffer, then test that you can decompress it, then test that it fails with the size limiter in place.

If that is too complex then you can probably stop here and not do the compression/decompression dance in the else part as that is not really testing anything specific to this patchset.

Unless you are just adding a new compression/decompression test because those were completely missing.
If that were the case though I would definitely get a test vector (otherwise you could get a valid match but fail to interoperate with any 3rd party) and separate this in a separate commit of its own

Reply:
Thanks @simo5 . You are right. I will try to change the test to limit to the decompression scenario. Let me please set the whole compression/decompression scenario, but just checking decompression limit